lib/router.js

```diff
@@ -90,22 +90,20 @@ Route.prototype.match = function (url, options) {
 
     // 2. check path
     // remove query string and hash fragment from url
-    var pos = url.indexOf('?');
-    var path;
-    if (pos >= 0) {
-        // remove query string
-        path = url.substring(0, pos);
-    } else {
-        pos = url.indexOf('#');
+    //
+    // hash fragment does not get sent to server.
+    // But since routr can be used on both server and client,
+    // we should remove hash fragment before matching the regex.
+    var path = url;
+    var pos;
+
+    // Leave `pos` at the beginning of the query-string, if any.
+    ['#', '?'].forEach(function(delimiter){
+        pos = url.indexOf(delimiter);
```

Comment:
thanks @gigabo for the fix! would it be better to use `pos = path.indexOf(delimeter)`?